### PR TITLE
Skip docker image creation if default master is used

### PR DESCRIPTION
**Pull request by kamilchodola**
## Changes

- Do not build master image if default config is used since this one is being built after each commit merged to master branch

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [X] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

#### Notes on testing

Tested on this run when `kch/optimize_oneClick` was also added to the check next to master branch - as expected it skipped two steps for triggering and waiting for docker image to be done
https://github.com/NethermindEth/nethermind/actions/runs/6089715555

Tested on this run where only master branch was checked (final version) - as expected it triggers build since it was triggered from `kch/optimize_oneClick` branch:
https://github.com/NethermindEth/nethermind/actions/runs/6089729660

#### Requires documentation update

- [ ] Yes
- [X] No

#### Requires explanation in Release Notes

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by stdevMac — APPROVED** [review]

File: .github/workflows/run-a-single-node-from-branch.yml [hunk 1]
```diff
@@ -94,7 +94,25 @@ jobs:
...
+        echo "REPO_NAME=$(echo $GITHUB_REPOSITORY | cut -d / -f 2)" >> $GITHUB_ENV
+        
+    - name: Check if master branch and default additional_options
+      id: check_conditions
+      run: |
+        if 
+        [[ "${{ github.ref }}" == "refs/heads/master" ]] && 
+        [[ "${{ steps.extract_dockerfile.outputs.dockerfile }}" == "Dockerfile" ]] && 
+        [[ "${{ steps.extract_dockerfile.outputs.build-config }}" == "release" ]]; then
+          echo "skip=true" >> $GITHUB_OUTPUT
```
Comment (brbrr) [inline .github/workflows/run-a-single-node-from-branch.yml:109]: It's relatively minor, but maybe we can make the variable name more explicit? Something like `skip_docker_build`

Comment (kamilchodola) [reply .github/workflows/run-a-single-node-from-branch.yml]: Good catch - applied change

File: .github/workflows/run-a-single-node-from-branch.yml [hunk 2]
```diff
@@ -94,7 +94,25 @@ jobs:
         echo "dockerfile=$(echo '${{ inputs.additional_options }}' | jq -r .default_dockerfile)" >> $GITHUB_OUTPUT
         echo "build-config=$(echo '${{ inputs.additional_options }}' | jq -r .default_dockerfile_build_type | tr '[:upper:]' '[:lower:]')" >> $GITHUB_OUTPUT
 
+    - name: Set Repo and Org Variables
+      run: |
+        echo "ORG_NAME=$(echo $GITHUB_REPOSITORY | cut -d / -f 1)" >> $GITHUB_ENV
+        echo "REPO_NAME=$(echo $GITHUB_REPOSITORY | cut -d / -f 2)" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (kamilchodola) [reply .github/workflows/run-a-single-node-from-branch.yml]: Yeah very nice - forgot abotu some stuff, also catched one variable which is not needed at all.

Applied changes and testing if it works properly

File: .github/workflows/run-a-single-node-from-branch.yml [hunk 3]
```diff
@@ -83,18 +83,32 @@ jobs:
...
       id: extract_dockerfile
       run: |
         echo "dockerfile=$(echo '${{ inputs.additional_options }}' | jq -r .default_dockerfile)" >> $GITHUB_OUTPUT
         echo "build-config=$(echo '${{ inputs.additional_options }}' | jq -r .default_dockerfile_build_type | tr '[:upper:]' '[:lower:]')" >> $GITHUB_OUTPUT
 
+    - name: Set Repo and Org Variables
+      run: |
+        echo "ORG_NAME=${{ github.repository_owner }}" >> $GITHUB_ENV
+        echo "REPO_NAME=${{ github.event.repository.name }}" >> $GITHUB_ENV
+
```
Comment (brbrr) [inline .github/workflows/run-a-single-node-from-branch.yml:97]:
```suggestion
```

File: .github/workflows/run-a-single-node-from-branch.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kamilchodola) [reply .github/workflows/run-a-single-node-from-branch.yml]: And just FYI - this is done this way becuase we have also one more repository which is direct fork of nethermind client repo and to make this workflow also work there we are not setting names explicitly in docker publish trigger but we have this reading of owner/repo and pushed as envs to trigger docker publish action

File: .github/workflows/run-a-single-node-from-branch.yml (hunk 3 above)
Comment (kamilchodola) [reply .github/workflows/run-a-single-node-from-branch.yml:97]: Seems to be working fine:
https://github.com/NethermindEth/nethermind/actions/runs/6095440360/job/16538991314

File: .github/workflows/run-a-single-node-from-branch.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (brbrr) [reply .github/workflows/run-a-single-node-from-branch.yml]: ah, ok. So this variables are actually used somewhere else. got it!

**Review by brbrr — APPROVED** [review]
LGTM

**Pull request merged**